Restructure MAAS doc (#1114)
spectrocloud/librarium: content/docs/04-clusters/02-data-center/01-maas/15-create-manage-maas-clusters.md

---
title: "Create and Manage MAAS Clusters"
metaTitle: "Create and Manage MAAS Clusters"
metaDescription: "Learn how to create and manage MAAS clusters in Palette."
hideToC: false
fullWidth: false
---

import Tabs from 'shared/components/ui/Tabs';
import WarningBox from 'shared/components/WarningBox';
import InfoBox from 'shared/components/InfoBox';
import PointsOfInterest from 'shared/components/common/PointOfInterest';

# Overview 

Palette supports creating and managing Kubernetes clusters deployed to a MAAS account. This section guides you on how to create a Kubernetes cluster in MAAS that is managed by Palette.

# Prerequisites

- An installed Private Cloud Gateway (PCG). Review [Install and Manage MAAS Gateway](/clusters/data-center/maas/install-manage-maas-pcg) for guidance.

- At least one preconfigured MAAS cloud account with credentials.

- A cluster profile for the MAAS environment. Review [Cluster Profiles](/cluster-profiles) for more information. 

# Deploy a MAAS cluster

To deploy a new MAAS cluster:

<br />

> 1. Log in to [Palette](https://console.spectrocloud.com) as a tenant admin.

Reviewer: or .. your palette instance if self hosted

2. Navigate to the **Main Menu** and click **Clusters**. Then click the **Add New Cluster** button.

3. Click **Deploy New Cluster** on the Create a New Cluster page.

4. Select **MAAS** and click the **Start MAAS Configuration** button.

5. Provide basic cluster information: **Cluster name**, **Description**, and **Tags**.

6. Select your cloud account from the **drop-down Menu** and click **Next**.

> 7. Select the cluster profile for your MAAS cluster. 

Reviewer: you will of course have needed to create a maas cluster profile first, and this should be a prerequisite (?)

8. Review and override pack parameters as desired and click **Next**. By default, parameters for all packs are set with values defined in the cluster profile.

> 9. Select a MAAS Cloud account and provide placement information.

Reviewer: not sure if Cloud should be capitalised

| Instance type | The compute instance type to be used for all nodes in the node pool. |
| Availability Zones | These are multiple, isolated locations within a Region. When you choose one or more availability zones, Palette provides fault tolerance to guard against hardware failures and network failures by provisioning nodes across availability zones if multiple zones are selected. |
| Disk Size | Specify the required storage size. |

12. You can configure the following cluster management features now if needed, or you can do it later:

- Manage Machines
- Scan Policies
- Backup Policies
- Review settings and deploy the cluster. 

13. Review settings and deploy the cluster. 

# Validation

You can validate your cluster is available by reviewing the cluster details page. Navigate to the left **Main Menu** and click **Clusters**. The **Clusters** page lists all available clusters Palette manages. Select the cluster to review its details page. Ensure the **Cluster Status** field contains the value **Running**.

# Delete a MAAS Cluster

> When you delete a MAAS cluster, all virtual machines and associated storage disks that were created for the cluster are removed. 

Reviewer: ```suggestion
When you delete a MAAS cluster, all machines and associated storage disks that were created for the cluster are removed. 
```

Follow these steps to delete a MAAS cluster.

<br />

1. Log in to [Palette](https://console.spectrocloud.com) as a tenant admin.

2. Navigate to the **Main Menu** and click **Clusters**. 

3. Select the cluster you want to delete.

4. Click the **Settings drop-down Menu**, and choose **Delete**.

> The cluster status is updated to Deleting while cluster resources are being deleted. When all resources are successfully deleted, the cluster status is updated to Deleted and the cluster is removed from the list. The delete operation returns the edge hosts to the **Ready** state. All the artifacts related to the Kubernetes distribution are removed.

Reviewer: is "Deleted" meant to be capitalised? since it is a status, shouldn't it be listed as code or highlighted in some other way? there is another example later in this paragraph

Author: Made **Deleted** and **Deleting** statuses boldface.

## Force Delete a Cluster

> If a cluster is stuck in the Deletion state for a minimum of 15 minutes it becomes eligible for force deletion. You can force delete a cluster from the tenant and project admin scope. To force delete a cluster follow the same steps outlined in [Delete a MAAS Cluster](/clusters/data-center/maas/create-manage-maas-clusters#deleteamaascluster). However, after 15 minutes, a **Force Delete Cluster** option is available in the **Settings drop-down menu**. The drop-down menu will provide you with an estimated time left before the force deletion becomes available.

Reviewer: ```suggestion
If a cluster is stuck in the Deletion state for a minimum of 15 minutes it becomes eligible for force deletion. You can force delete a cluster from the tenant and project admin scope. To force delete a cluster follow the same steps outlined in [Delete a MAAS Cluster](/clusters/data-center/maas/create-manage-maas-clusters#deleteamaascluster). After 15 minutes, a **Force Delete Cluster** option is available in the **Settings drop-down menu**. The drop-down menu will provide you with an estimated remaining duration before the force deletion becomes available.
```

<br />

<WarningBox>

> A force delete can result in Palette-provisioned resources to be missed in the removal process. Verify there are no remaining resources. Failure to remove provisioned resources can result in unexpected costs.

Reviewer: ```suggestion
A force delete can result in Palette-provisioned resources being missed during the removal process. Verify there are no remaining resources. Failure to remove provisioned resources can result in unexpected costs.
```

Author: fixed

</WarningBox>

# Next Steps

> Now that you’ve deployed a MAAS cluster, you can start developing and deploying applications to your cluster. We recommend you review the Day-2 responsibilities and become familiar with the cluster management tasks. Check out the [Manage Clusters](/clusters/cluster-management) documentation to learn more about Day-2 responsibilities.

Reviewer: not sure what Day-2 "responsibilities" means. Do we mean "operations" perhaps?

Author: @antongisli yes, this should be "operations". This was from the original doc. Thanks for catching it.

| Allow worker capability | Select this option to allow workloads to be provisioned on master nodes. |
| [Labels](/clusters/cluster-management/taints#overviewonlabels) | Labels apply placement constraints on a pod. For example, you can add a label to make a node eligible to receive the workload. |
| [Taints](/clusters/cluster-management/taints#overviewontaints) | Sets toleration to pods and allows (but does not require) the pods to schedule onto nodes with matching taints. |
| Instance type | The compute instance type to be used for all nodes in the node pool. |
| Availability Zones | These are multiple, isolated locations within a Region. When you choose one or more availability zones, Palette provides fault tolerance to guard against hardware failures and network failures by provisioning nodes across availability zones if multiple zones are selected. |
| Disk Size | Specify the required storage size. |

<br />

#### Worker Node Pool

| Property | Description |
|-----------|-------------|
| Name | A descriptive name for the node pool. |
| Size | Number of VMs to be provisioned for the node pool.|

> | Rolling Update | Rolling update has two available options. Review the Update Parameter list below for more details. ?? |

Reviewer 2: There is no Update Parameter list below. This needs attention.

2. Navigate to the **Main Menu** and click **Clusters**. 

3. Select the cluster you want to delete.

4. Click the **Settings drop-down Menu**, and choose **Delete**.

The cluster status is updated to **Deleting** while cluster resources are being deleted. When all resources are successfully deleted, the cluster status is updated to **Deleted** and the cluster is removed from the list. The delete operation returns the edge hosts to the **Ready** state. All the artifacts related to the Kubernetes distribution are removed.

## Force Delete a Cluster

If a cluster is stuck in the Deletion state for a minimum of 15 minutes it becomes eligible for force deletion. You can force delete a cluster from the tenant and project admin scopes. To force delete a cluster, follow the same steps outlined in [Delete a MAAS Cluster](/clusters/data-center/maas/create-manage-maas-clusters#deleteamaascluster). After 15 minutes, a **Force Delete Cluster** option is available in the **Settings drop-down menu**. The drop-down menu will provide you with an estimated remaining duration before the force deletion becomes available.

<br />

<WarningBox>

> A force delete can result in Palette-provisioned resources being missed during the removal process. Verify there are no remaining resources. Failure to remove provisioned resources can result in unexpected costs.

Reviewer 2: "Verify there are no remaining resources" should be clarified to include reference to MAAS.

Maybe, "Verify that any MAAS machines associated with the cluster have been released"

---
title: "Create and Manage MAAS Clusters"
metaTitle: "Create and Manage MAAS Clusters"
metaDescription: "Learn how to create and manage MAAS clusters in Palette."
hideToC: false
fullWidth: false
---

import Tabs from 'shared/components/ui/Tabs';
import WarningBox from 'shared/components/WarningBox';
import InfoBox from 'shared/components/InfoBox';
import PointsOfInterest from 'shared/components/common/PointOfInterest';

# Overview 

Palette supports creating and managing Kubernetes clusters deployed to a MAAS account. This section guides you on how to create a Kubernetes cluster in MAAS that is managed by Palette.

# Prerequisites

> - An installed Private Cloud Gateway (PCG).

Reviewer 3: ```suggestion
- An installed Private Cloud Gateway (PCG). Review the [Install and Manage MAAS Gateway](/clusters/data-center/maas/install-manage-maas-pcg) for guidance.
```

- At least one preconfigured MAAS cloud account with credentials.

- A cluster profile for the MAAS environment. Review [Cluster Profiles](/cluster-profiles) for more information. 

# Enablement

> `video: title: "maas-cluster-creation": ./cluster-creation-videos/maas.mp4`

Reviewer 3: Let's remove this video.

Author: video removed

To deploy a new MAAS cluster:

<br />

1. Log in to [Palette](https://console.spectrocloud.com) as a tenant admin.

> 2. Navigate to the **Main** menu and click **Clusters**. Then click the **Add New Cluster** button.

Reviewer 3: ```suggestion
2. Navigate to the **Main Menu** and click **Clusters**. Then click the **Add New Cluster** button.
```

3. Click **Deploy New Cluster** on the Create a New Cluster page.

4. Select **MAAS** and click the **Start MAAS Configuration** button.

5. Provide basic cluster information: **Cluster name**, **Description**, and **Tags**.

> 6. Select your cloud account from the drop-down menu and click **Next**.

Reviewer 3: ```suggestion
6. Select your cloud account from the **drop-down Menu** and click **Next**.
```

7. Select the cluster profile for your MAAS cluster. 

8. Review and override pack parameters as desired and click **Next**. By default, parameters for all packs are set with values defined in the cluster profile.

9. Select a MAAS Cloud account and provide placement information.

> 10. Select a domain from the **Domain drop-down menu** and click **Next**. 

Reviewer 3: ```suggestion
10. Select a domain from the **Domain drop-down Menu** and click **Next**. 
```

| [Labels](/clusters/cluster-management/taints#overviewonlabels) | Labels apply placement constraints on a pod. For example, you can add a label to make a node eligible to receive the workload. |
| [Taints](/clusters/cluster-management/taints#overviewontaints) | Sets toleration to pods and allows (but does not require) the pods to schedule onto nodes with matching taints. |
| Instance type | The compute instance type to be used for all nodes in the node pool. |
| Availability Zones | These are multiple, isolated locations within a Region. When you choose one or more availability zones, Palette provides fault tolerance to guard against hardware failures and network failures by provisioning nodes across availability zones if multiple zones are selected. |
| Disk Size | Specify the required storage size. |

12. You can configure the following cluster management features now if needed, or you can do it later:

- Manage Machines
- Scan Policies
- Backup Policies
- Review settings and deploy the cluster. 

13. Review settings and deploy the cluster. 

# Validation

> You can validate your cluster is up and running by reviewing the cluster details page. Navigate to the left **Main Menu** and click **Clusters**. The **Clusters** page contains a list of all available clusters Palette manages. Select the cluster to review its details page. Ensure the **Cluster Status** field contains the value **Running**.

Reviewer 3: ```suggestion
You can validate your cluster is available by reviewing the cluster details page. Navigate to the left **Main Menu** and click **Clusters**. The **Clusters** page lists all available clusters Palette manages. Select the cluster to review its details page. Ensure the **Cluster Status** field contains the value **Running**.
```

# Delete a MAAS Cluster

When you delete a MAAS cluster, all virtual machines and associated storage disks that were created for the cluster are removed. 

Follow these steps to delete a MAAS cluster.

<br />

1. Log in to [Palette](https://console.spectrocloud.com) as a tenant admin.

> 2. Navigate to the **Main** menu and click **Clusters**. 

Reviewer 3: ```suggestion
2. Navigate to the **Main Menu** and click **Clusters**. 
```

3. Select the cluster you want to delete.

4. Click the **Settings drop-down Menu**, and choose **Delete**.

> The cluster status is updated to Deleting while cluster resources are being deleted. When all resources are successfully deleted, the cluster status is updated to Deleted and the cluster is removed from the list. The delete operation returns the edge hosts to the Ready state. All the artifacts related to the Kubernetes distribution are removed.

Reviewer 3: ```suggestion
The cluster status is updated to Deleting while cluster resources are being deleted. When all resources are successfully deleted, the cluster status is updated to Deleted and the cluster is removed from the list. The delete operation returns the edge hosts to the **Ready** state. All the artifacts related to the Kubernetes distribution are removed.
```

| Allow worker capability | Select this option to allow workloads to be provisioned on master nodes. |
| [Labels](/clusters/cluster-management/taints#overviewonlabels) | Labels apply placement constraints on a pod. For example, you can add a label to make a node eligible to receive the workload. |
| [Taints](/clusters/cluster-management/taints#overviewontaints) | Sets toleration to pods and allows (but does not require) the pods to schedule onto nodes with matching taints. |
| Instance type | The compute instance type to be used for all nodes in the node pool. |
| Availability Zones | These are multiple, isolated locations within a Region. When you choose one or more availability zones, Palette provides fault tolerance to guard against hardware failures and network failures by provisioning nodes across availability zones if multiple zones are selected. |
| Disk Size | Specify the required storage size. |

<br />

#### Worker Node Pool

| Property | Description |
|-----------|-------------|
| Name | A descriptive name for the node pool. |
| Size | Number of VMs to be provisioned for the node pool.|

> | Rolling Update | Rolling update has two available options. Review the Update Parameter list below for more details.|

Reviewer 3: @ritawatson let's break this update list out to its own table like we have done for other sub-settings before.

Author: @karl-cardenas-coding tracked down the info and created a table for these.

---
title: "Create and Manage MAAS Clusters"
metaTitle: "Create and Manage MAAS Clusters"
metaDescription: "Learn how to create and manage MAAS clusters in Palette."
hideToC: false
fullWidth: false
---

import Tabs from 'shared/components/ui/Tabs';
import WarningBox from 'shared/components/WarningBox';
import InfoBox from 'shared/components/InfoBox';
import PointsOfInterest from 'shared/components/common/PointOfInterest';

# Overview 

Palette supports creating and managing Kubernetes clusters deployed to a MAAS account. This section guides you on how to create a Kubernetes cluster in MAAS that is managed by Palette.

# Prerequisites

- An installed Private Cloud Gateway (PCG). Review [Install and Manage MAAS Gateway](/clusters/data-center/maas/install-manage-maas-pcg) for guidance.

> - At least one preconfigured MAAS cloud account with credentials.

Reviewer 3: ```suggestion
- A MAAS account registered in Palette. Refer to the [Register and Manage MAAS Cloud Accounts](/clusters/data-center/maas/register-manage-maas-cloud-accounts) if you need to register a MAAS account in Palette.
```
We use the term register in the how-to guide, so we should use the same wording here. This is also a good opportunity to cross-link.

| [Taints](/clusters/cluster-management/taints#overviewontaints) | Sets toleration to pods and allows (but does not require) the pods to schedule onto nodes with matching taints. |
| Cloud configuration | Set the Resource Pool and Availability Zone in MAAS to select available servers from for deployment. Filter available servers to only those that have at least the amount of CPU and Memory selected. |
| Resource pool | The MAAS resource pool from which to select available servers deployment.  |
| Availability Zones | The MAAS Availability Zones from which to select available servers for deployment. If you select multiple zones, Palette will deploy servers evenly across them as long as sufficient servers are available to do so. |

<br />

#### Worker Node Pool

| Property | Description |
|-----------|-------------|
| Node pool name | A descriptive name for the node pool. |
| Number of nodes in the pool | Number of nodes to be provisioned for the node pool.|
| Additional [labels](/clusters/cluster-management/taints#overviewonlabels) | Optional labels apply placement constraints on a pod. For example, you can add a label to make a node eligible to receive the workload. |

> | [Taints](/clusters/cluster-management/taints#overviewontaints) | Sets toleration to pods and allows (but does not require) the pods to schedule onto nodes with matching taints. |

Reviewer 3: @ritawatson both these sections are documented in the cluster management [Node Pools page](https://docs.spectrocloud.com/clusters/cluster-management/node-pool).  Your call, but in the other providers, I have added verbiage and linked to this page.

Author: @karl-cardenas-coding I didn't realize that. I'll check verbiage in docs for other providers and refer.

Author: Removed tables and added link to Node Pool page.

| Disk Size | Specify the required storage size. |

<br />

#### Worker Node Pool

| Property | Description |
|-----------|-------------|
| Node pool name | A descriptive name for the node pool. |
| Number of nodes in the pool | Number of VMs to be provisioned for the node pool.|
| Additional [Labels](/clusters/cluster-management/taints#overviewonlabels) | Optional labels apply placement constraints on a pod. For example, you can add a label to make a node eligible to receive the workload. |
| [Taints](/clusters/cluster-management/taints#overviewontaints) | Sets toleration to pods and allows (but does not require) the pods to schedule onto nodes with matching taints. |
| Rolling update | There are two available options: **Expand First** and **Contract First**. Review descriptions of these options below for more details.|
| Instance type | The compute instance type to be used for all nodes in the node pool. |
| Managed disk | Select a disk and specify the required storage size. |

> | Availability zones | These are multiple, isolated locations within a Region. When you choose one or more availability zones, Palette provides fault tolerance to guard against hardware failures and network failures by provisioning nodes across availability zones if multiple zones are selected. |

Reviewer 4: Change description to: The MAAS Availability Zone(s) to select available servers from for deployment. If you select multiple zones, Palette will deploy servers evenly across them as long as sufficient servers are available to do so.

#### Rolling Update Options for Node Pools

| Property | Description |
|-----------|-------------|
| Expand First | Palette launches a new node first, then shuts down the old node before providing updates. |
| Contract First | Palette shuts down the old node first, then launches a new node before providing updates.|

12. You can configure the following cluster management features now if needed, or you can do it later:

    - Manage Machines
    - Scan Policies
    - Backup Policies
    - Review settings and deploy the cluster. 

> 

Reviewer 4: Change to:
    - Manage Machines
    - Schedule Scans
    - Schedule Backups
    - RBAC
    - Location
